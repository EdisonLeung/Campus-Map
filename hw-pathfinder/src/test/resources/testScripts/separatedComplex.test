# Tests pathfinding on a graph where 2 sets of nodes are isolated
# from each other

# Create Graph
CreateGraph separated

# Add nodes
AddNode separated a
AddNode separated b
AddNode separated c
AddNode separated d
AddNode separated e
AddNode separated f

# Add edges
AddEdge separated a b 10.0
AddEdge separated a c 1.0
AddEdge separated b a 3.0
AddEdge separated c b 4.0
AddEdge separated d f 1.0
AddEdge separated f e 3.0
AddEdge separated e d 2.0

# test pathfinding
FindPath separated a b

FindPath separated b a

FindPath separated c a

FindPath separated d e

FindPath separated b d

FindPath separated f a

FindPath separated a e