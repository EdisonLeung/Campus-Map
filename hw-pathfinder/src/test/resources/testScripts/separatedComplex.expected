# Tests pathfinding on a graph where 2 sets of nodes are isolated
# from each other

# Create Graph
created graph separated

# Add nodes
added node a to separated
added node b to separated
added node c to separated
added node d to separated
added node e to separated
added node f to separated

# Add edges
added edge 10.000 from a to b in separated
added edge 1.000 from a to c in separated
added edge 3.000 from b to a in separated
added edge 4.000 from c to b in separated
added edge 1.000 from d to f in separated
added edge 3.000 from f to e in separated
added edge 2.000 from e to d in separated

# test pathfinding
path from a to b:
a to c with weight 1.000
c to b with weight 4.000
total cost: 5.000

path from b to a:
b to a with weight 3.000
total cost: 3.000

path from c to a:
c to b with weight 4.000
b to a with weight 3.000
total cost: 7.000

path from d to e:
d to f with weight 1.000
f to e with weight 3.000
total cost: 4.000

path from b to d:
no path found

path from f to a:
no path found

path from a to e:
no path found